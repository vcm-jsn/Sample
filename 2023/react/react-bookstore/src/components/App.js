//import logo from "./logo.svg";
import { Component } from "react";
import Header from "./Header.js";
import Main from "./Main.js";
import Footer from "./Footer.js";
import "./App.css";
import productsData from "../data/products";
//import { useState } from "react";
import { useState, useEffect } from "react";

function App() {
  // constructor(props) {
  //   super(props);
  //   // this.state = {
  //   //   itemsInCart: ["10", "14", "16"],
  //   //   firstName: "Siri",
  //   // };
  //   this.state = {
  //     itemsInCart: [],
  //     products: [],
  //     loading: false,
  //     firstName: "Siri",
  //   };
  //   this.addToCart = this.addToCart.bind(this);
  //   this.removeFromCart = this.removeFromCart.bind(this);
  // }
  const [itemsInCart, setItemsInCart] = useState([]);
  const [products, setProducts] = useState([]);
  const [isLoading, setIsLoading] = useState(false);
  const firstName = "Siri";

  useEffect(() => {
    async function fetchData() {
      try {
        setIsLoading(true);
        const response = await fetch("/data/products.json");
        const json = await response.json();
        shuffleArray(json);
        setProducts(json);
        setIsLoading(false);
      } catch (e) {
        console.error(e);
      }
    }
    fetchData();
  }, []);

  // componentDidMount() {
  //   this.setState({ loading: true });
  //   fetch("http://localhost:3000/data/products.json")
  //     .then((response) => response.json())
  //     .then((products) => this.shuffleArray(products))
  //     .then((products) => {
  //       this.setState({ products: products, loading: false });
  //     });
  // }

  function shuffleArray(array) {
    for (let i = array.length - 1; i > 0; i--) {
      let j = Math.floor(Math.random() * (i + 1));
      let temp = array[i];
      array[i] = array[j];
      array[j] = temp;
    }
    return array;
  }

  // function addToCart(id) {
  //   let newItems = [...this.state.itemsInCart, id];
  //   this.setState({
  //     itemsInCart: newItems,
  //   });
  // }
  // function removeFromCart(idToRemove) {
  //   let newItems = this.state.itemsInCart.filter((id) => id !== idToRemove);
  //   this.setState({ itemsInCart: newItems });
  // }

  function addToCart(id) {
    let newItems = [...itemsInCart, id];
    setItemsInCart(newItems);
  }

  function removeFromCart(idToRemove) {
    let newItems = itemsInCart.filter((id) => id !== idToRemove);
    setItemsInCart(newItems);
  }
  if (isLoading) {
    return <p>Loading...</p>;
  }
  //let firstName = "Siri";
  //const [itemsInCart, setItemsInCart] = useState(["10", "14", "16", "4"]);

  return (
    <div data-testid="app" className="container">
      <Header firstName={firstName} />
      <Main
        products={products}
        itemsInCart={itemsInCart}
        setItemsInCart={setItemsInCart}
        addToCart={addToCart}
        removeFromCart={removeFromCart}
      />
      <Footer firstName={firstName} />
    </div>
  );
}

export default App;
