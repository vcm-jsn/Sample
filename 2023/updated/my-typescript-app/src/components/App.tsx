//import logo from "./logo.svg";
import { Component } from "react";
import Header from "./Header.js";
import Main from "./Main";
import Footer from "./Footer.js";
import "./App.css";
//import productsData from "../data/products";
//import { useState } from "react";
import { useState, useEffect } from "react";
import Book from "./Book";

function App() {
  const [itemsInCart, setItemsInCart] = useState<Array<string>>([]);
  const [products, setProducts] = useState<Array<Book>>([]);
  const [isLoading, setIsLoading] = useState<Boolean>(false);
  const firstName = "Siri";

  useEffect(() => {
    async function fetchData() {
      try {
        setIsLoading(true);
        const response = await fetch(
          "http://localhost:3000/data/products.json"
        );
        const json = await response.json();
        setProducts(shuffleArray(json));
        setIsLoading(false);
      } catch (e) {
        console.error(e);
      }
    }
    fetchData();
  }, [setProducts]);

  function shuffleArray(array: Book[]) {
    for (let i = array.length - 1; i > 0; i--) {
      let j = Math.floor(Math.random() * (i + 1));
      let temp = array[i];
      array[i] = array[j];
      array[j] = temp;
    }
    return array;
  }

  // function addToCart(id) {
  //   let newItems = [...this.state.itemsInCart, id];
  //   this.setState({
  //     itemsInCart: newItems,
  //   });
  // }
  // function removeFromCart(idToRemove) {
  //   let newItems = this.state.itemsInCart.filter((id) => id !== idToRemove);
  //   this.setState({ itemsInCart: newItems });
  // }

  function addToCart(id: string) {
    let newItems = [...itemsInCart, id];
    setItemsInCart(newItems);
  }

  function removeFromCart(idToRemove: string) {
    let newItems = itemsInCart.filter((id) => id !== idToRemove);
    setItemsInCart(newItems);
  }
  if (isLoading) {
    return <p>Loading...</p>;
  }

  return (
    <div data-testid="app" className="container">
      <Header firstName={firstName} />
      <Main
        products={products}
        itemsInCart={itemsInCart}
        //setItemsInCart={setItemsInCart}
        addToCart={addToCart}
        removeFromCart={removeFromCart}
      />
      <Footer firstName={firstName} />
    </div>
  );
}

export default App;
