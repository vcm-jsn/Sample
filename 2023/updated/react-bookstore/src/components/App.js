//import logo from "./logo.svg";
import { Component } from "react";
import Header from "./Header.js";
import Main from "./Main.js";
import Footer from "./Footer.js";
import "./App.css";
import productsData from "../data/products";
//import { useState } from "react";
import { useState, useEffect } from "react";
import * as actionCreators from "../actions";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";

function App(props) {
  // constructor(props) {
  //   super(props);
  //   // this.state = {
  //   //   itemsInCart: ["10", "14", "16"],
  //   //   firstName: "Siri",
  //   // };
  //   this.state = {
  //     itemsInCart: [],
  //     products: [],
  //     loading: false,
  //     firstName: "Siri",
  //   };
  //   this.addToCart = this.addToCart.bind(this);
  //   this.removeFromCart = this.removeFromCart.bind(this);
  // }
  // const [itemsInCart, setItemsInCart] = useState([]);
  // const [products, setProducts] = useState([]);
  // const [isLoading, setIsLoading] = useState(false);
  const [isLoading, setIsLoading] = useState(false);
  const [shuffledProductList, setShuffledProductList] = useState([]);
  const firstName = "Siri";
  const {
    products,
    itemsInCart,
    loadProducts,
    addToCart,
    removeFromCart,
    submitCart,
    readCart,
  } = props;

  // useEffect(() => {
  //   async function fetchData() {
  //     try {
  //       setIsLoading(true);
  //       const response = await fetch("/data/products.json");
  //       const json = await response.json();
  //       shuffleArray(json);
  //       setProducts(json);
  //       setIsLoading(false);
  //     } catch (e) {
  //       console.error(e);
  //     }
  //   }
  //   fetchData();
  // }, []);

  useEffect(() => {
    async function fetchData() {
      try {
        setIsLoading(true);
        const response = await fetch(
          "http://localhost:3000/data/products.json"
        );
        const json = await response.json();

        loadProducts(json);
        readCart();
        setIsLoading(false);
      } catch (e) {
        console.error(e);
      }
    }
    fetchData();
  }, [loadProducts, readCart]);

  // componentDidMount() {
  //   this.setState({ loading: true });
  //   fetch("http://localhost:3000/data/products.json")
  //     .then((response) => response.json())
  //     .then((products) => this.shuffleArray(products))
  //     .then((products) => {
  //       this.setState({ products: products, loading: false });
  //     });
  // }

  useEffect(() => {
    let shuffled = shuffleArray(props.products);
    setShuffledProductList(shuffled);
  }, [props.products]);

  function shuffleArray(array) {
    for (let i = array.length - 1; i > 0; i--) {
      let j = Math.floor(Math.random() * (i + 1));
      let temp = array[i];
      array[i] = array[j];
      array[j] = temp;
    }
    return array;
  }

  // function addToCart(id) {
  //   let newItems = [...this.state.itemsInCart, id];
  //   this.setState({
  //     itemsInCart: newItems,
  //   });
  // }
  // function removeFromCart(idToRemove) {
  //   let newItems = this.state.itemsInCart.filter((id) => id !== idToRemove);
  //   this.setState({ itemsInCart: newItems });
  // }

  // function addToCart(id) {
  //   let newItems = [...itemsInCart, id];
  //   setItemsInCart(newItems);
  // }

  // function removeFromCart(idToRemove) {
  //   let newItems = itemsInCart.filter((id) => id !== idToRemove);
  //   setItemsInCart(newItems);
  // }
  // if (isLoading) {
  //   return <p>Loading...</p>;
  // }
  //let firstName = "Siri";
  //const [itemsInCart, setItemsInCart] = useState(["10", "14", "16", "4"]);

  return (
    <div data-testid="app" className="container">
      <Header firstName={firstName} />
      <Main
        products={products}
        itemsInCart={itemsInCart}
        addToCart={addToCart}
        removeFromCart={removeFromCart}
        submitCart={submitCart}
      />
      {/* <Main
        products={products}
        itemsInCart={itemsInCart}
        setItemsInCart={setItemsInCart}
        addToCart={addToCart}
        removeFromCart={removeFromCart}
      /> */}
      <Footer firstName={firstName} />
    </div>
  );
}
const mapStateToProps = (state, props) => {
  return {
    itemsInCart: state.cart.items,
    products: state.products.products,
  };
};
const mapDispatchToProps = (dispatch) => {
  return bindActionCreators(actionCreators, dispatch);
};

export default connect(mapStateToProps, mapDispatchToProps)(App);

//export default App;
