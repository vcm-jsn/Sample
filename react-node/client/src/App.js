import React , {Component, useState} from 'react';
import logo from './logo.svg';
import './App.css';
import Navbar from './components/Navbar'
import Background from './components/Background'
import Sidebar from './components/Sidebar'
import {
  BrowserRouter as Router,
  Route
} from  'react-router-dom'
import Employees from './components/Employees'
import Home from './components/Home'
import EmployeeDetails from './components/EmployeeDetails'


function App() {

  const routes = [
    {
      path:'/',
      exact:true,
      main:() =><Home/>
    },
    {
      path:'/getEmployees',
      exact:true,
      main:() =><Employees/>
    },
    {
      path:'employeeDetails',
      exact:true,
      main:() =><EmployeeDetails/>
    },
    {
      path:'/test',
      exact:true,
      main:() =><h1> Test Page</h1>
    },

  ]
  const [isOpened, setIsOpened] = useState(false);
  return (
        <>
        <Router>
        <div style ={{flex:1, padding :'90px'}}>

        {routes.map((route, index)=> (
        
        <Route 
          key={index}
          path={route.path}
          exact={route.exact}
          component={route.main}
          />
          ))}

        </div>

        </Router>
   <Background setIsOpened = {setIsOpened} show={isOpened}/>
    <Navbar toggleMenu={setIsOpened}/>
    <Sidebar show ={isOpened} setIsOpened={setIsOpened}/>
   </>
  )
}

export default App;
