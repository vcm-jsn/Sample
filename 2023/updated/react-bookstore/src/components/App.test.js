import { render, screen, fireEvent } from "@testing-library/react";
import App from "./App";

// test("renders learn react link", () => {
//   render(<App />);
//   const cont = screen.getByTestId("app");
//   expect(cont).toBeInTheDocument();
//   //expect(linkElement).toBeInTheDocument();
// });

describe("<App />", () => {
  it("makes the api call", () => {
    jest.mock("../data/products");
    render(<App />);
    expect(screen.getByText("Loading")).toBeInTheDocument();
  });

  it("renders testing text", () => {
    render(<App />);
    const testText = screen.getByText(/Learning React, SIRI/i);
    expect(testText).toBeInTheDocument();
  });

  it("renders as expected", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  test("clicking the button toggles button messages", () => {
    render(<App />);
    let buttonText;

    setTimeout(() => {
      const buttons = screen.getAllByText(/Add to Cart/i);
      fireEvent.click(buttons[0]);
      buttonText = buttons[0].getByText(/In Cart/i);
      expect(buttonText).toBeInDocument();

      fireEvent.click(buttons[0]);
      buttonText = buttons[0].getByText(/Add to Cart/i);
      expect(buttonText).toBeInDocument();
    }, 2000);
  });
});
