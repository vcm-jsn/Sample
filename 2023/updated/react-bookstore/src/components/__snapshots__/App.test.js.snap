// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders as expected 1`] = `
<div>
  <p>
    Loading...
  </p>
</div>
`;
